exporter/cortexexporter/cortex.go

```diff
@@ -0,0 +1,354 @@
+// quantile of every data point as a Sample, and adding each Sample to its corresponding TimeSeries.
+// tsMap and metric cannot be nil.
+func (ce *cortexExporter) handleSummaryMetric(tsMap map[string]*prompb.TimeSeries, metric *otlp.Metric) error {
+
+	if metric.SummaryDataPoints == nil {
+		return fmt.Errorf("invalid metric type: wants summary points")
+	}
+
+	for _, pt := range metric.SummaryDataPoints {
+
+		time := int64(pt.GetTimeUnixNano())
+		mType := metric.GetMetricDescriptor().GetType()
+
+		// sum and count of the Summary should append suffix to baseName
+		baseName := getPromMetricName(metric.GetMetricDescriptor(), ce.namespace)
+
+		// treat sum as sample in an individual TimeSeries
+		sum := &prompb.Sample{
+			Value:     pt.GetSum(),
+			Timestamp: time,
+		}
+		sumLbs := createLabelSet(pt.GetLabels(), nameStr, baseName+sumStr)
+		addSample(tsMap, sum, sumLbs, mType)
+
+		// treat count as a sample in an individual TimeSeries
+		count := &prompb.Sample{
+			Value:     float64(pt.GetCount()),
+			Timestamp: time,
+		}
+		countLbs := createLabelSet(pt.GetLabels(), nameStr, baseName+countStr)
+		addSample(tsMap, count, countLbs, mType)
+
+		// process each percentile/quantile
+		for _, qt := range pt.GetPercentileValues() {
+			quantile := &prompb.Sample{
+				Value:     float64(qt.Value),
+				Timestamp: time,
+			}
+			percentileStr := strconv.FormatFloat(qt.Percentile, 'f', -1, 64)
+			qtLbs := createLabelSet(pt.GetLabels(), nameStr, baseName, quantileStr, percentileStr)
+			addSample(tsMap, quantile, qtLbs, mType)
+		}
+	}
+	return nil
+}
+
+// Because we are adhering closely to the Remote Write API, we must Export a
+// Snappy-compressed WriteRequest instance of the TimeSeries Metrics in order
+// for the Remote Write Endpoint to properly receive our Metrics data.
+func (ce *cortexExporter) export(ctx context.Context, TsMap map[string]*prompb.TimeSeries) error {
+	//Calls the helper function to convert the TsMap to the desired format
+	req, err := wrapTimeSeries(TsMap)
+	if err != nil {
+		return err
+	}
+
+	//Uses proto.Marshal to convert the WriteRequest into wire format (bytes array)
+	data, err := proto.Marshal(req)
+	if err != nil {
+		return err
+	}
+
+	//Makes use of the snappy compressor, as we are emulating the Remote Write package
+	compressedData := snappy.Encode(nil, data)
+
+	//Create the HTTP POST request to send to the endpoint
+	httpReq, err := http.NewRequest("POST", ce.endpoint, bytes.NewReader(compressedData))
+	if err != nil {
+		return err
+	}
+
+	//Add optional headers
+	for name, value := range ce.headers {
+		httpReq.Header.Set(name, value)
+	}
+
+	//Add necessary headers: https://cortexmetrics.io/docs/apis/#remote-api
+	httpReq.Header.Add("Content-Encoding", "snappy")
+	httpReq.Header.Set("Content-Type", "application/x-protobuf")
+	httpReq.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
```

Comment:
Should this be hard coded?

Reply:
As of right now, it seems as though these currrent parameter headers are necessary to interact with the Corte gateway: https://cortexmetrics.io/docs/apis/#remote-api
Would you recommend us trying to dynamically change this to new versions as they come out-- if they come out?